modules/shared/src/main/scala/org/tessellation/schema/trust.scala

```diff
@@ -1,13 +1,61 @@
 package org.tessellation.schema
 
+import cats.syntax.either._
+import cats.syntax.order._
+
+import scala.util.control.NoStackTrace
+
 import org.tessellation.schema.peer.PeerId
 
-import derevo.cats.show
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
 
 object trust {
 
+  @derive(eqv, show)
+  case class TrustDbValues(
+    peerId: PeerId,
+    score: Option[Double],
+    rating: Option[Double],
+    observationAdjustment: Option[Double]
+  ) {
+    protected def add(value: Option[Double], that: Option[Double]): Either[ValueArithmeticError, Option[Double]] = {
+      val sum = value.flatMap(x => that.map(x + _))
+
+      if (sum >= value && sum >= that) {
+        Option[Double](sum.get).asRight[ValueArithmeticError]
+      } else ValueOverflow.asLeft[Option[Double]]
+    }
+
+    def addToScore(that: Option[Double]): Either[ValueArithmeticError, TrustDbValues] = {
+      val newScore = add(score, that)
+
+      newScore match {
+        case Left(_)      => ValueOverflow.asLeft[TrustDbValues]
+        case Right(value) => TrustDbValues(peerId, value, rating, observationAdjustment).asRight[ValueArithmeticError]
+      }
+    }
+
+    def addToRating(that: Option[Double]): Either[ValueArithmeticError, TrustDbValues] = {
+      val newScore = add(rating, that)
+
+      newScore match {
+        case Left(_)      => ValueOverflow.asLeft[TrustDbValues]
+        case Right(value) => TrustDbValues(peerId, score, value, observationAdjustment).asRight[ValueArithmeticError]
+      }
+    }
+
+    def addToObservationAdjustment(that: Option[Double]): Either[ValueArithmeticError, TrustDbValues] = {
+      val newScore = add(observationAdjustment, that)
+
+      newScore match {
+        case Left(_)      => ValueOverflow.asLeft[TrustDbValues]
+        case Right(value) => TrustDbValues(peerId, score, rating, value).asRight[ValueArithmeticError]
+      }
+    }
+  }
```

Comment:
please take a look at monocle, it can simplify that code

Reply:
Incorporated monocle.

Reply:
`monocle` is no longer needed.